sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala

```diff
@@ -20,47 +20,38 @@ package com.intel.hibench.sparkbench.ml
 
 import org.apache.spark.{SparkConf, SparkContext}
 
-import org.apache.spark.mllib.classification.LogisticRegressionWithLBFGS
-import org.apache.spark.mllib.regression.LabeledPoint
-import org.apache.spark.rdd.RDD
+import org.apache.spark.ml.classification.LogisticRegression
+import org.apache.spark.ml.feature.LabeledPoint
+import org.apache.spark.sql.SparkSession
 
 object LogisticRegression {
 
   def main(args: Array[String]): Unit = {
-    var inputPath = ""
-
-    if (args.length == 1) {
-      inputPath = args(0)
-    }
-
-    val conf = new SparkConf()
-	.setAppName("LogisticRegressionWithLBFGS")
-    val sc = new SparkContext(conf)
-
-    // $example on$
-    // Load training data in LIBSVM format.
-    val data: RDD[LabeledPoint] = sc.objectFile(inputPath)
-
-    // Split data into training (60%) and test (40%).
-    val splits = data.randomSplit(Array(0.6, 0.4), seed = 11L)
-    val training = splits(0).cache()
-    val test = splits(1)
-
+    val inputPath = args(0)
+    val numFeatures = args(1).toInt
+    val aggDepth = if (args.length >2) {
+        if (args(2).toInt < 2) 2 else args(2).toInt
+    } else 0
+ //  val sqlContext= new org.apache.spark.sql.SQLContext(sc)
+//   import sqlContext.implicits._
+   val spark = SparkSession.builder().appName("LogisticRegression").getOrCreate()
+   val df = spark.read.parquet(args(0))
+/*
+    val spark = SparkSession.builder.appName("LogisticRegression").getOrCreate()
+    val df = spark.read.format("libsvm")
+                       .option("numFeatures", numFeatures)
+                       .load(inputPath)
+*/
     // Run training algorithm to build the model
-    val model = new LogisticRegressionWithLBFGS()
-      .setNumClasses(10)
-      .run(training)
-
-    // Compute raw scores on the test set.
-    val predictionAndLabels = test.map { case LabeledPoint(label, features) =>
-      val prediction = model.predict(features)
-      (prediction, label)
-    }
-
-    val accuracy = predictionAndLabels.filter(x => x._1 == x._2).count().toDouble / predictionAndLabels.count()
-    println(s"Accuracy = $accuracy")
-
-    sc.stop()
+    val model = new LogisticRegression()
+      .setMaxIter(30)
+      .setRegParam(0.3)
+      .setElasticNetParam(0.8)
+// uncomment below config for spark 2.1 or above
```

Comment:
is there any solution to enable both Spark 2.1 and 2.2? we cannot reply on end user will change the code by following your comments in the code.

Reply:
currently I cannot think of any solution to support this change, suggest removing this change.

Reply:
`if (spark.SPARK_VERSION.startWith("2.2" ) {
..
} else if (..) {
}
`